students/donggyu/users/views.py

```diff
@@ -1,3 +1,51 @@
-from django.shortcuts import render
+import json
+import re
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+
+from users.models import User
+
+class UserView(View):
+    def post(self, request):
+        # 유저 회원가입
+        try:
+            data = json.loads(request.body)
+
+            post_name         = data['name']
+            post_email        = data['email']
+            post_password     = data['password']
+            post_phone_number = data['phone_number']
+
+            # 이메일 데이터 유효성 검사
+            cheak_email = re.match('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$', post_email)
+
+            if cheak_email == None:
+                return JsonResponse({"message" : "EMAIL_DATA_ERROR"}, status = 400)
+
+            # 패스워드 데이터 유효성 검사
+            cheak_password = re.match('^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$', post_password)
+
+            if cheak_password == None:
+                return JsonResponse({"message" : "PASSWORD_DATA_ERROR"}, status = 400)
```

Comment:
re.match 의 type은 bool type입니다. 즉, True or False를 반환하는데 
현재와 같은 조건문이라면 원하시는 기능을 하지 못할 것 같습니다. (False 와 None은 다르기 때문입니다.)

아래의 코드 블럭을 참고하여 수정해주세요!
```suggestion
            REGEX_EMAIL    = ...
            REGEX_PASSWORD = ...
            
            if not re.match(REGEX_EMAIL, email):
                return JsonResponse({"message" : "EMAIL_VALIDATE_ERROR"}, status = 400)
            
            if not re.match(REGEX_PASSWORD, password):
                return JsonResponse({"message" : "PASSWORD_VALIDATE_ERROR"}, status = 400)

```

Reply:
코드는 멘토님이 바꾸어 주신 코드로 수정하였습니다.
그런데 궁금한 점이 있습니다.

![스크린샷, 2022-07-08 19-09-16](https://user-images.githubusercontent.com/98671555/177988507-e234f3bf-5dd2-44c6-b053-9aab56b022fa.png)
![스크린샷, 2022-07-08 19-08-52](https://user-images.githubusercontent.com/98671555/177989422-2f140764-1914-4d08-9e5b-9f1a054285ff.png)
![스크린샷, 2022-07-08 19-09-06](https://user-images.githubusercontent.com/98671555/177989472-8d0aaadc-e39e-490c-94fe-b598558f889d.png)

첫 번째 이미지는 제가 쉘을 통해 반환되는 타입을 확인하기 위해 테스트 해본 이미지입니다. 제가 보기에는 True, False가 아닌 객체를 반환하거나 None을 반환하거나 하는 것으로 보입니다.

두, 세번째 이미지는 제가 기존의 코드로 했을때 조건문이 제 기능을 하는 모습을 캡처한 것입니다. 혹시 이 부분에 대해서 제가 잘못알고 있는 부분이 있는지 알고 싶습니다.

Reply:
리뷰에 혼동이 있었네요 shell로 확인해보신 점 너무 훌륭합니다!
동규님이 직접 확인 하신 것처럼  re.match는 객체를 반환하고 정규식에 맞지 않는 값과 비교했을 때는 None을 반환하기에 기존 코드로도 충분히 제기능을 할 수 있습니다.